File: November 2022/20 November 2022/Java/LCM Triplet.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            long N = Long.parseLong(read.readLine());
            Solution ob = new Solution();
            System.out.println(ob.lcmTriplets(N));
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution {
    long lcmTriplets(long N) {
        // code here
        if(N == 1){
            return 1;
        }
        if(N == 2){
            return 2;
        }
        if(N%2 != 0){
            return N*(N-1)*(N-2);
        }
        
        if(N%3 == 0){
            return (N-1)* (N-2)*(N-3);
        }
        return Math.max(N*(N-1) * (N-3), (N-1)* (N-2)*(N-3));
    }
}

File: November 2022/20 November 2022/C++/LCM Triplet.cpp
//{ Driver Code Starts
// Initial Template for C++

// Initial Template for C++
// Back-end complete function Template for C++
// User function Template for C++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution {
  public:
    long long lcmTriplets(long long N) {
        if(N<=2)   
        return N;
     else if(N%2==0 && N%3==0)
        return (N-1)*(N-2)*(N-3);
     else if(N%2==0)
        return N*(N-1)*(N-3);
     return N*(N-1)*(N-2);
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        long long N;
        cin >> N;
        Solution ob;
        cout << ob.lcmTriplets(N) << "\n";
    }
}
// } Driver Code Ends